### ISSUE TITLE
About model on test set of tusimple

### ISSUE THREAD

**Issue by Cverlpeng** [wvangansbeke/LaneDetection_End2End#22]
thanks you  work!
I try to run model on the test set of tusimple ,I get good result!!!
but I can't show gt on image , I don't know how to get beta params of gt lane.
I tried to convert the lane coordinates of the test set to the top view, then fit the lane on the top view and get the curve parameters twice, but when I use the obtained parameters to draw the line onto the picture, I get the wrong result, may I ask? Do I need a line to normalize the coordinates and then fit?

```python
def draw_fitted_line(img, params, resize, color=(255,0,0)):
    params = params.data.cpu().tolist()
    y_stop = 0.7
    y_prime = np.linspace(0, y_stop, 20)
    params = [0] * (4 - len(params)) + params
    d, a, b, c = [*params]
    x_pred = d*(y_prime**3) + a*(y_prime)**2 + b*(y_prime) + c
    x_pred = x_pred*(2*resize-1)
    y_prime = (1-y_prime)*(resize-1)
    lane = [(xcord, ycord) for (xcord, ycord) in zip(x_pred, y_prime)] 
    img = cv2.polylines(img, [np.int32(lane)], isClosed = False, color = color,thickness = 1)
    return img, lane
```

the y_prime is normalize the coordinates ? so beta parms of  model output is normalize the coordinates ?
best!!
thanks

**Comment by wvangansbeke** [comment]
Hi @Cverlpeng,

You are correct. The coordinates need to be mapped to a bird's eye view perspective before you preform the line fitting. I normalized the coordinates to make it more robust for training, i.e. less sensitive to hyperparameter tuning. The code above,  takes into account that the coefficients are normalized for a grid between [0 - 1]. So you just need to normalize your values with the aspect ratio of the original image, `(2*resize, resize)`.

Kind regards.

**Comment by Cverlpeng** [comment]
I get it 
thanks!!!
